examples/event_handling/EventHandling.cpp

```diff
@@ -0,0 +1,338 @@
+
+                        if constexpr (std::is_same_v<T, sf::Event::Closed>)
+                        {
+                            m_window.close();
+                        }
+                        else if constexpr (std::is_same_v<T, sf::Event::KeyPressed>)
+                        {
+                            m_log.emplace_back("Key Pressed: " + sf::Keyboard::getDescription(event.scancode));
+
+                            // When the enter key is pressed, switch to the next handler type
+                            if (event.code == sf::Keyboard::Key::Enter)
+                            {
+                                m_handlerType = HandlerType::Forward;
+                                m_handlerText.setString("Current Handler: Forward");
+                            }
+                        }
+                        else if constexpr (std::is_same_v<T, sf::Event::MouseMoved>)
+                        {
+                            m_log.emplace_back("Mouse Moved: " + vec2ToString(event.position));
+                        }
+                        else if constexpr (std::is_same_v<T, sf::Event::MouseButtonPressed>)
+                        {
+                            m_log.emplace_back("Mouse Pressed");
+                        }
+                        else if constexpr (std::is_same_v<T, sf::Event::TouchBegan>)
+                        {
+                            m_log.emplace_back("Touch Began: " + vec2ToString(event.position));
+                        }
+                        else
+                        {
+                            // All unhandled events will end up here
+                            // m_log.emplace_back("Other Event");
+                        }
+                    });
+            }
+            else if (m_handlerType == HandlerType::Forward)
+            {
+                // Forward to other callable
+                // In this case we forward it to our handle member functions
+                // we defined for the concrete event types we want to handle
+                // When choosing this method a default "catch-all" handler must
+                // be available for unhandled events to be forwarded to
+                // If you don't want to provide an empty "catch-all" handler
+                // you will have to make sure (e.g. via if constexpr) that this
+                // lambda doesn't attempt to call a member function that doesn't exist
+                m_window.handleEvents([this](const auto& event) { handle(event); });
+            }
+
+            // Limit the log to 24 entries
+            if (m_log.size() > 24u)
+                m_log.erase(m_log.begin(), m_log.begin() + static_cast<int>(m_log.size() - 24u));
+
+            // Draw the contents of the log to the window
+            m_window.clear();
+
+            for (std::size_t i = 0; i < m_log.size(); ++i)
+            {
+                m_logText.setPosition({50.f, static_cast<float>(i * 20) + 50.f});
```

Comment:
`i * 20.f` would remove the need for a cast

Reply:
This is weird... thought the compiler would have preferred promoting `std::size_t * float` to `double` instead, I guess the standard doesn't mind that precision is lost here.

Reply:
I mean if it starts erroring with loss in conversion, then I guess we need this.

Reply:
The current code works with a cast and is guaranteed not to produce any conversion warnings so...